### PR TITLE
clippy fixes

### PR DESCRIPTION
**Pull request by andrewmwells-amazon**
## Description of changes
Automatically applied clippy fixes

## Checklist for requesting a review

The change in this PR is (choose one, and delete the other options):

- [x] A change "invisible" to users (e.g., documentation, changes to "internal" crates like `cedar-policy-core`, `cedar-validator`, etc.)

I confirm that this PR (choose one, and delete the other options):

- [x] Does not update the CHANGELOG because my change does not significantly impact released code.

I confirm that [`cedar-spec`](https://github.com/cedar-policy/cedar-spec) (choose one, and delete the other options):

- [x] Does not require updates because my change does not impact the Cedar Dafny model or DRT infrastructure.

## Disclaimer

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by john-h-kastner-aws — APPROVED** [review]

File: cedar-policy-core/src/est.rs [hunk 1]
```diff
@@ -356,7 +356,7 @@ mod test {
         let ast = est
             .try_into_ast_policy(None)
             .expect("Failed to convert to AST");
-        ast.try_into().expect("Failed to convert to EST")
```
Comment (andrewmwells-amazon) [inline cedar-policy-core/src/est.rs:359]: Not sure if we actually want this change.

Comment (cdisselkoen) [reply cedar-policy-core/src/est.rs:359]: I think this is fine.  Probably the AST->EST transform was originally fallible but now is not (as we can see from there now being an `Into` impl)?

**Review by cdisselkoen — APPROVED** [review]

File: cedar-policy-core/src/est.rs (hunk 1 above)
Comment (john-h-kastner-aws) [reply cedar-policy-core/src/est.rs:359]: Seems legit. We have `impl From<ast::Policy> for Policy`, so the `TryFrom` should be automatic from that

**Comment by andrewmwells-amazon** [comment]
It seems the CedarJava failure is caused by this change: https://github.com/cedar-policy/cedar/actions/runs/7465002067/job/20313236263?pr=577

**Comment by andrewmwells-amazon** [comment]
Nvm, the issue is a property test that will fail non-deterministically (as Victor noticed here: https://github.com/cedar-policy/cedar-java/pull/72#discussion_r1437156141). Unrelated to this PR.

**Review by john-h-kastner-aws — APPROVED** [review]

**Pull request merged**